core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -353,6 +354,9 @@ static private void setupNodeDirectories(File baseDirectory,
     private final TestKitNodes nodes;
     private final Map<Integer, ControllerServer> controllers;
     private final Map<Integer, BrokerServer> brokers;
+
+    private final Map<Integer, Boolean> aliveBrokers;
```

Comment:
Instead of tracing alive brokers, maybe we should add a new method `isShutdown` to `KafkaBroker`?

Reply:
Sounds great but if a broker already shutdown, how can we access the server and check its status?

Reply:
> if a broker already shutdown, how can we access the server and check its status?

both zk and kraft broker has in-memory state, so maybe you can leverage them?